### PR TITLE
Move out tracking to a util and send the branch name to segment

### PR DESCRIPTION
**Pull request by Iamsamwhoami**

### CODE REVIEW

File: src/utils/track-event.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const trackEvent = (eventName, options) => {
+  if (window && window.analytics) {
+    window.analytics.track(eventName, {
+      ...options,
+      branch: process.env.BRANCH,
```
Comment (Iamsamwhoami) [inline src/utils/track-event.js]: Will this work? will the BRANCH env variable be there compile time?

Comment (hedvigoscar) [reply src/utils/track-event.js]: 🤷‍♂️

**Comment by Iamsamwhoami** [comment]
@hedvigoscar this works now 🎉

**Comment by hedvigoscar** [comment]
🙌

File: plugins/svgr/gatsby-node.js [hunk 2]
```diff
@@ -24,5 +26,17 @@ exports.modifyWebpackConfig = ({ config }, options) => {
     ],
   });
 
+  config.merge({
+    plugins: [
+      new webpack.DefinePlugin({
+        NETLIFY_BRANCH: JSON.stringify(
+          process.env.BRANCH ? process.env.BRANCH : 'local',
```
Comment (hedvigoscar) [inline plugins/svgr/gatsby-node.js]: maybe `process.node.ENV || 'local'`, not really important though

**Comment by Iamsamwhoami** [comment]
@hedvigoscar hehe I realised that I had put the webpack config into the svgr plugin 🙈created a new separate plugin just for this now! Also fixed the thing you commented on!

**Comment by hedvigoscar** [comment]
Actually make sure that the linter is happy before you merge :+1:

**Comment by Iamsamwhoami** [comment]
Woops 🙈

**Pull request merged**